mix/qml/js/Debugger.js

```diff
@@ -30,9 +36,59 @@ function init(data)
 	currentDisplayedState = 0;
 	setupInstructions(currentSelectedState);
 	setupCallData(currentSelectedState);
-	statesSlider.maximumValue = data.states.length - 1;
+	initLocations();
+	initSlider();
+	selectState(currentSelectedState);
+}
+
+function updateMode()
+{
+	initSlider();
+}
+
+function initLocations()
+{
+	locations = [];
+	if (debugData.states.length === 0)
+		return;
+
+	var nullLocation = { start: -1, end: -1, documentId: "", state: 0 };
+	var prevLocation = nullLocation;
+
+	for (var i = 0; i < debugData.states.length - 1; i++) {
+		var code = debugData.states[i].code;
+		var location = (code.documentId !== "") ? code.locations[codeStr(i)] : nullLocation;
```

Comment:
Executing code which is not part of the project (e.g. standard contract, or imported from blockchain)